sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PathFilterSuite.scala

```diff
@@ -0,0 +1,557 @@
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class PathFilterSuite
+    extends QueryTest
+    with SharedSparkSession
+    with AdaptiveSparkPlanHelper {
+  import testImplicits._
+
+  test("SPARK-31962: when modifiedAfter specified with a past date") {
+    withTempDir { dir =>
+      withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
+        val path = new Path(dir.getCanonicalPath)
+        val file = new File(dir, "file1.csv")
+        stringToFile(file, "text")
+        file.setLastModified(DateTimeUtils.currentTimestamp())
+        val df = spark.read
+          .option("modifiedAfter", "2019-05-10T01:11:00")
+          .format("csv")
+          .load(path.toString)
+        assert(df.count() == 1)
+      }
+    }
+  }
+
+  test("SPARK-31962: when modifiedBefore specified with a future date") {
+    withTempDir { dir =>
+      withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
+        val path = new Path(dir.getCanonicalPath)
+        val file = new File(dir, "file1.csv")
+        stringToFile(file, "text")
+        val df = spark.read
+          .option("modifiedBefore", "2090-05-10T01:11:00")
+          .format("csv")
+          .load(path.toString)
+        assert(df.count() == 1)
+      }
+    }
+  }
+
+ test("SPARK-31962: when modifiedBefore specified with a past date") {
+    withTempDir { dir =>
+      withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
+        val path = new Path(dir.getCanonicalPath)
+        val file = new File(dir, "file1.csv")
+        stringToFile(file, "text")
+        file.setLastModified(DateTimeUtils.currentTimestamp())
+        val msg = intercept[AnalysisException] {
+          spark.read
+            .option("modifiedBefore", "1984-05-01T01:00:00")
+            .format("csv")
+            .load(path.toString)
+        }.getMessage
+        assert(msg.contains("Unable to infer schema for CSV"))
+      }
+    }
+  }
+
+    test("SPARK-31962: when modifiedAfter specified with a past date, " +
+        "multiple files, one valid") {
+        withTempDir { dir =>
+            withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
+                val path = new Path(dir.getCanonicalPath)
+                val file1 = new File(dir, "file1.csv")
+                val file2 = new File(dir, "file2.csv")
+                stringToFile(file1, "text")
+                stringToFile(file2, "text")
+                file1.setLastModified(DateTimeUtils.currentTimestamp())
+                file2.setLastModified(0)
+                val df = spark.read
+                    .option("modifiedAfter", "2019-05-10T01:11:00")
+                    .format("csv")
+                    .load(path.toString)
+                assert(df.count() == 1)
+            }
+        }
+    }
+    test("SPARK-31962: when modifiedAfter specified with a past date, " +
+        "multiple files, both valid") {
```

Comment:
What's a difference between this test and https://github.com/apache/spark/pull/28841/files#diff-9bb59882dc9a25fa8355a712f8ca964eR38 ? I think the two test has the same code path, so only this test looks enough.

Reply:
I updated the descriptions and distinguished the tests to be more clear.